clustpy/deep/neural_networks/_abstract_autoencoder.py

```diff
@@ -181,10 +182,42 @@ def loss(self, batch: list, loss_fn: torch.nn.modules.loss._Loss, device: torch.
         batch_data = batch[1].to(device)
         embedded = self.encode(batch_data)
         reconstructed = self.decode(embedded)
-        loss = loss_fn(reconstructed, batch_data)
+        loss = ssl_loss_fn(reconstructed, batch_data)
         return loss, embedded, reconstructed
 
-    def evaluate(self, dataloader: torch.utils.data.DataLoader, loss_fn: torch.nn.modules.loss._Loss,
+    def loss_augmentation(self, batch: list, ssl_loss_fn: torch.nn.modules.loss._Loss, device: torch.device) -> (
+            torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor):
+        """
+        Calculate the loss of a single batch of data and an augmented version of the data.
+        Note that the augmented samples come at position batch[1] and the original samples at batch[2].
+
+        Parameters
+        ----------
+        batch : list
+            the different parts of a dataloader (id, augmented samples, original samples, ...)
+        ssl_loss_fn : torch.nn.modules.loss._Loss
+            self-supervised learning (ssl) loss function for training the network, e.g. reconstruction loss
+        device : torch.device
+            device to be trained on
+
+        Returns
+        -------
+        loss : (torch.Tensor, torch.Tensor, torch.Tensor)
+            the combined network loss of the sample and the augmented sample,
+            the embedded input sample,
+            the reconstructed input sample,
+            the embedded augmented sample,
+            the reconstructed augmented sample
+        """
+        assert type(batch) is list, "batch must come from a dataloader and therefore be of type list"
+        batches_orig = [batch[i] for i in range(2, len(batch), 2)]
```

Comment:
why do you need the list comprehension here? couldn't you just use:

`batches_orig = batch[2]`

maybe add a comment to clarify this further

Reply:
added a comment

Reply:
I see, makes sense :)